salt/cloud/clouds/ec2.py

```diff
@@ -4136,11 +4136,14 @@ def create_keypair(kwargs=None, call=None):
 
     data = aws.query(params,
                      return_url=True,
+                     return_root=True,
                      location=get_location(),
                      provider=get_provider(),
                      opts=__opts__,
                      sigver='4')
-    return data
+    keys = [ x for x in data[0] if 'requestId' not in x ]
```

Comment:
can we do a check to make sure that we get the correct https status code from the query before we try to parse the data?

Also, this is going to have some lint warnings, you will need to remove the extra spaces between `[x` and `x]`

Reply:
I am sorry. This is something I did a year ago. don't have time to re-setup the env to fix and push again. 
feel free to update yourself, don't worry about the pr credit

Reply:
Ok, i will just push them to your repo once I get a chance to make the changes